Gems/PhysX/Code/Source/RigidBodyComponent.h

```diff
@@ -148,14 +153,17 @@ namespace PhysX
     private:
         void SetupConfiguration();
         void CreatePhysics();
+        void ApplyPhysxSpecificConfiguration();
         void InitPhysicsTickHandler();
         void PostPhysicsTick(float fixedDeltaTime);
 
         const AzPhysics::RigidBody* GetRigidBodyConst() const;
 
         std::unique_ptr<TransformForwardTimeInterpolator> m_interpolator;
 
-        AzPhysics::RigidBodyConfiguration m_configuration;
+        AzPhysics::RigidBodyConfiguration m_configuration; //!< Generic properties from AzPhysics.
+        RigidBodyConfiguration
+            m_physxSpecificConfiguration; //!< Properties specific to PhysX which might not have exact equivalents in other physics engines.
```

Comment:
Instead of having 2 configuration classes and members, would it be possible to have only one by making the `PhysX::RigidBodyConfiguration` class extend `AzPhysics::RigidBodyConfiguration` to add the physx specific properties? I thought that was the way to extend the physics generic framework. Seems like having to handle 2 classes independently for configuration properties makes it more complicated.

Reply:
I tried that first, but it was a complete nightmare. I also actually quite like having the 2 separate classes, because if you don't want to set any of the advanced properties you can just not bother about the specific configuration.

Reply:
Not a big deal, I only mentioned it because this introduces a different pattern. It's fine to make the separation of advanced settings, that's nice for the user, I wonder if that separation could be done at serialization without having to have 2 structures. In any case, this works.